sqlx-core/src/postgres/types/mod.rs

```diff
@@ -205,6 +205,9 @@ mod json;
 #[cfg(feature = "ipnetwork")]
 mod ipnetwork;
 
+#[cfg(feature = "ipnetwork")]
```

Comment:
If you change this to not use `ipnetwork` then it can be enabled by default.

I'd also add it to this module: https://github.com/launchbadge/sqlx/blob/master/sqlx-macros/src/database/postgres.rs#L66

after the `ipnetwork` declarations so as to not be a breaking change (if the `ipnetwork` feature is enabled it will continue to use those types).

Reply:
I'd rather not reimplement the parsing, and `IpAddr` is not exposing the subnet mask, it's not a natural type for `inet` or `cidr`.

Reply:
Yeah, that occurred to me at the same time as you said it. 

I think it's going to be a source of confusion if `Decode for IpAddr` just throws away the subnet mask, though; it should either be an error to decode a value with a subnet mask smaller than the full width of the address, or just don't implement `Decode` at all and only implement `Encode`.

Reply:
If the latter, since the encoding part is relatively straightforward I'd still prefer if it didn't depend on `ipnetwork`.

Reply:
Or, if we provide a type like `PgInet` that is a pair of `IpAddr` and `u8` and build on that, that might be better.

Reply:
It's a bit useless without `Decode`.

I don't think throwing away the subnet is a bad idea, it's why you're using `IpAddr` in the first place.

I think it's important to be explicit with the support for this because once you enable ipnetwork you'll hopefully check why it's not on by default and find out about subnets.

Reply:
I disagree, I think people will just see that it requires the `ipnetwork` feature and enable it without thinking about it. Or it'll already be enabled because they're using the `all-types` feature. There's a reason why we didn't just do this in the first place.

Reply:
The most I can accept is returning an error if the subnet is non-zero, in general I don't think it should be an error because you're already doing a lossy conversion, it's an explicit and known lossy conversion, otherwise you'd be using `IpNetwork` in the first place.

I do not want the subnet, and I do not want to use `IpNetwork` in my public structs because there is no subnet and it complicates some things.